view/frontend/web/instantsearch.js

```diff
@@ -207,408 +206,454 @@ requirejs(['algoliaBundle','Magento_Catalog/js/price-utils'], function(algoliaBu
-					item: $('#instant-hit-template').html()
+					item: $('#instant-hit-template').html(),
 				},
-				transformData: {
-					item: function (hit) {
+				transformItems: function (items) {
+					return items.map(function (item) {
+						var hit = Object.assign({}, item);
 						hit = transformHit(hit, algoliaConfig.priceKey, search.helper);
+						// FIXME: transformHit is a global
 						hit.isAddToCartEnabled = algoliaConfig.instant.isAddToCartEnabled;
-						
 						hit.algoliaConfig = window.algoliaConfig;
-						
-						var state = search.helper.state;
-						hit.__position = (state.page * state.hitsPerPage) + hit.__hitIndex + 1;
-						
 						return hit;
-					}
+					})
 				}
 			};
-			
+
 			/**
-			 * Pagination
-			 * Docs: https://community.algolia.com/instantsearch.js/v2/widgets/pagination.html
+			 * pagination
+			 * Docs: https://www.algolia.com/doc/api-reference/widgets/pagination/js/
 			 **/
 			allWidgetConfiguration.pagination = {
 				container: '#instant-search-pagination-container',
-				cssClass: 'algolia-pagination',
-				showFirstLast: false,
-				maxPages: 1000,
-				labels: {
+				showFirst: false,
+				showLast: false,
+				showNext: true,
+				showPrevious: true,
+				totalPages: 1000,
+				templates: {
 					previous: algoliaConfig.translations.previousPage,
 					next: algoliaConfig.translations.nextPage
 				},
-				scrollTo: 'body'
 			};
-			
+
 			delete allWidgetConfiguration.infiniteHits;
 		}
-		
+
 		/**
 		 * Here are specified custom attributes widgets which require special code to run properly
-		 * Custom widgets can be added to this object like [attributeName]: function(facet, templates)
-		 * Function must return instantsearch.widget object
-		 * Docs: https://community.algolia.com/instantsearch.js/v2/widgets.html
+		 * Custom widgets can be added to this object like [attribute]: function(facet, templates)
+		 * Function must return an array [<widget name>: string, <widget options>: object]
 		 **/
 		var customAttributeFacet = {
 			categories: function (facet, templates) {
 				var hierarchical_levels = [];
 				for (var l = 0; l < 10; l++)
 					hierarchical_levels.push('categories.level' + l.toString());
-				
+
 				var hierarchicalMenuParams = {
 					container: facet.wrapper.appendChild(createISWidgetContainer(facet.attribute)),
 					attributes: hierarchical_levels,
 					separator: ' /// ',
+					templates: templates,
 					alwaysGetRootLevel: true,
 					limit: algoliaConfig.maxValuesPerFacet,
-					templates: templates,
-					sortBy: ['name:asc'],
-					cssClasses: {
-						list: 'hierarchical',
-						root: 'facet hierarchical'
-					}
+					sortBy: ['name:asc']
 				};
-				
+
 				hierarchicalMenuParams.templates.item = '' +
-					'<div class="ais-hierearchical-link-wrapper">' +
-					'<a class="{{cssClasses.link}}" href="{{url}}">{{label}}' +
+					'<a class="{{cssClasses.link}} {{#isRefined}}{{cssClasses.link}}--selected{{/isRefined}}" href="{{url}}">{{label}}' + ' ' +
```

Comment:
I simply simplified this, I think adding the div was initially for styling, but we're going with a more simple/vanilla style.
In fact I would be even in favour of removing the template altogether. The default one is good enough imho
see https://instantsearchjs.netlify.com/stories/?path=/story/hierarchicalmenu--default